src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1024,7 +1047,7 @@ impl LinkCollector<'_, '_> {
                         return None;
                     }
                     res = prim;
-                    fragment = Some(path.to_owned());
+                    fragment = Some((*path.as_str()).to_owned());
```

Comment:
Unfortunately, that `as_str()` returns a `SymbolString` so this manual defer is required

Reply:
`SymbolStr` derefs to `str`, I think it should work without.

Reply:
`Some(path.as_str().to_owned())` did not. I'll try look for something else

Reply:
Can you post the error? If it doesn't work, it doesn't work but I'd like to know why.

Reply:
```
1018 |                     fragment = Some((path.as_str()).to_owned());
     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `std::string::String`, found struct `SymbolStr`
     |
help: try using a conversion method
     |
1018 |                     fragment = Some((path.as_str()).to_owned().to_string());
```

`to_string()` instead of `to_owned()` works as it should.